src/csharp/Grpc.IntegrationTesting/ExternalDnsWithTracingClientServerTest.cs

```diff
@@ -0,0 +1,171 @@
+#region Copyright notice and license
+
+// Copyright 2015 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Net.Sockets;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Grpc.Core;
+using Grpc.Core.Logging;
+using Grpc.Core.Utils;
+using Grpc.Testing;
+using NUnit.Framework;
+
+namespace Grpc.IntegrationTesting
+{
+    /// <summary>
+    /// See https://github.com/grpc/issues/18074, this test is meant to
+    /// try to trigger the described bug.
+    /// Runs interop tests in-process, with that client using a target
+    /// name that using a target name that triggers interaction with
+    /// external DNS servers (even though it resolves to the in-proc server).
+    /// </summary>
+    public class ExternalDnsWithTracingClientServerTest
+    {
+        Server server;
+        Channel channel;
+        TestService.TestServiceClient client;
+        SocketUsingLogger newLogger;
+
+        [OneTimeSetUp]
+        public void Init()
+        {
+            Environment.SetEnvironmentVariable("GRPC_TRACE", "all");
```

Comment:
I'm not sure if setting like this is going to work because of https://github.com/grpc/grpc/issues/14963

Reply:
Removed the sanity check and added a TODO (below comment)